elastalert/util.py

```diff
@@ -361,8 +361,11 @@ def build_es_conn_config(conf):
     parsed_conf['aws_region'] = None
     parsed_conf['profile'] = None
     parsed_conf['headers'] = None
-    parsed_conf['es_host'] = os.environ.get('ES_HOST', conf['es_host'])
-    parsed_conf['es_port'] = int(os.environ.get('ES_PORT', conf['es_port']))
+    es_host = os.environ.get('ES_HOST', conf['es_host'])
+    es_port = int(os.environ.get('ES_PORT', conf['es_port']))
+    parsed_conf['es_host'] = parse_host(es_host, es_port)
```

Comment:
Why is `es_host` getting the port appended to it and being converted to a list?  Pretty sure this will break AWS auth:

https://github.com/jertel/elastalert2/blob/27deb8a61dd48798c4686ec95d3e48909903a694/elastalert/util.py#L332

Reply:
When you say "will break AWS auth" are you referring to OpenSearch?

Reply:
If I'm reading the change correctly, `es_host` will become 

```
["{host}:{port}".format(host=host, port=port)]
```

From https://github.com/jertel/elastalert2/pull/548/files#diff-c94be3c6634086358da8944c7febe9c9e98fc1b4425fcc4e116e0740f5cf4eb6R556

This would break AWS auth because the host is used in the signing process.

Reply:
Sure looks like it would break it:

https://github.com/jertel/elastalert2/blob/80b7c4f24562954c03011e4a49dafaabe310398d/elastalert/util.py#L325

https://github.com/jertel/elastalert2/blob/80b7c4f24562954c03011e4a49dafaabe310398d/elastalert/util.py#L332

Reply:
This is the code that I think helps explain your argument better: https://github.com/DavidMuller/aws-requests-auth/blob/2e1dd0f37e3815c417c3b0630215a77aab5af617/aws_requests_auth/aws_auth.py#L118

Based on that, I agree. However, we get this request for supporting multiple ES hosts every month or so, so perhaps there's something we can do for the community to make this easier without the need for a load balancer. Would a safer way to do this be to introduce a new config parameter called `es_hosts` for those who want to use multiple ES hosts, outside of AWS? `util.py::build_es_conn_config()` could prefer that parameter if specified, and it would override the `es_host` parameter.

Reply:
I think an array over a csv string makes more sense anyway.

Reply:
Ok i can redo